tests/cpp/test_multidevice_overlap.cpp

```diff
@@ -196,36 +211,35 @@ class OverlapTest : public MultiDeviceTest {
 // clang-format on
 TEST_F(OverlapTest, ReduceScatterBasedPipeliningATenImplementation) {
   std::vector<c10::cuda::CUDAStream> streams;
-  for (auto j : c10::irange(params.S)) {
-    // define the sliced tensors
-    auto ta_j = ta_.select(0, j);
-    auto tc_locally_reduced_j = tc_locally_reduced_.select(0, j);
-    auto tc_j = tc_.select(0, j);
-
-    if (params.use_different_streams) {
-      auto new_stream = c10::cuda::getStreamFromPool(
-          /*isHighPriority=*/false, my_device_index_);
-      streams.push_back(new_stream);
-      setCurrentCUDAStream(new_stream);
+  std::generate_n(
+      std::back_inserter(streams),
+      params.number_of_streams,
```

Comment:
AFAICS from https://github.com/NVIDIA/Fuser/blob/346e51c285f2d511cc2a9455d46dd0c97f5787d2/tests/cpp/test_multidevice_overlap.cpp#L52, `params` is fixed to the default value. Therefore, the test never explores other values unless someone manually changes the source code? 

(Not a big deal; I'm just verifying my understanding because my first impression was "why not TEST_P")

Reply:
> the test never explores other values unless someone manually changes the source code?

That is correct, for now